src/Microsoft.Azure.EventHubs.ServiceFabricProcessor/EventProcessorService.cs

```diff
@@ -0,0 +1,488 @@
+                    }
+                }
+                if (receiver == null)
+                {
+                    EventProcessorEventSource.Current.Message("Out of retries creating receiver");
+                    throw new Exception("Out of retries creating event hub receiver", lastException);
+                }
+
+                //
+                // Instantiate user's event processor class and call Open.
+                // If user's Open code fails, treat that as a fatal exception and let it throw out.
+                //
+                EventProcessorEventSource.Current.Message("Creating event processor");
+                this.userEventProcessor = this.EventProcessorFactory.CreateEventProcessor(this.linkedCancellationToken, this.partitionContext);
+                await this.userEventProcessor.OpenAsync(this.linkedCancellationToken, this.partitionContext);
+                EventProcessorEventSource.Current.Message("Event processor created and opened OK");
+
+                //
+                // Start metrics reporting. This runs as a separate background thread.
+                //
+                Thread t = new Thread(this.MetricsHandler);
+                t.Start();
+
+                //
+                // Receive pump.
+                //
+                EventProcessorEventSource.Current.Message("RunAsync setting handler and waiting");
+                this.MaxBatchSize = this.Options.MaxBatchSize;
+                receiver.SetReceiveHandler(this, Options.InvokeProcessorAfterReceiveTimeout);
+                this.linkedCancellationToken.WaitHandle.WaitOne();
+
+                EventProcessorEventSource.Current.Message("RunAsync continuing, cleanup");
+            }
+            finally
+            {
+                if (this.userEventProcessor != null)
+                {
+                    // partitionContext is set up before processor is created, so it is available if processor is not null
+                    await this.userEventProcessor.CloseAsync(this.partitionContext, this.linkedCancellationToken.IsCancellationRequested ? CloseReason.Cancelled : CloseReason.Failure);
+                }
+                if (receiver != null)
+                {
+                    receiver.SetReceiveHandler(null);
+                    await receiver.CloseAsync();
+                }
+                if (ehclient != null)
+                {
+                    await ehclient.CloseAsync();
+                }
+                if (this.internalFatalException != null)
+                {
+                    throw this.internalFatalException;
+                }
+            }
+        }
+
+        /// <summary>
+        /// From IPartitionReceiveHandler
+        /// </summary>
+        public int MaxBatchSize { get; set; }
+
+        async Task IPartitionReceiveHandler.ProcessEventsAsync(IEnumerable<EventData> events)
+        {
+            if ((events != null) || ((events == null) && Options.InvokeProcessorAfterReceiveTimeout))
```

Comment:
>if ((events != null) || ((events == null) && Options.InvokeProcessorAfterReceiveTimeout)) [](start = 12, length = 89)

Do you need this since receive handler will handle this condition? Just create an empty event set if events==null.

Reply:
You are correct, no need to double-check since InvokeProcessorAfterReceiveTimeout is already hooked up to client receive pump.